OrcScala/src/orc/API.scala

```diff
@@ -48,7 +54,8 @@ trait OrcCompiler[+E] extends OrcCompilerProvides[E] with OrcCompilerRequires
 /** The interface from a caller to an Orc runtime
   */
 trait OrcRuntimeProvides {
-  @throws(classOf[ExecutionException])
+  // TODO: Remove the expression argument on run. It is tied to the token interpreter since other interperters do not use Nameless.
```

Comment:
Hmm. Let me look into that. I think the newer Backend interface subsumes some of this. And even if it does not. it would make sense to use the same type parameters as provided to the backend.